File: myBlogSystem/hero-parent/hero-web/src/main/webapp/js/service/blogtypeService.js
//服务层
app.service('blogtypeService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../blogtype/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../blogtype/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../blogtype/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../blogtype/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../blogtype/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../blogtype/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../blogtype/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: myBlogSystem/hero-parent/hero-interface/src/main/java/club/hero/service/BlogtypeService.java
package club.hero.service;
import java.util.List;
import club.hero.pojo.Blogtype;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface BlogtypeService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Blogtype> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Blogtype blogtype);
	
	
	/**
	 * 修改
	 */
	public void update(Blogtype blogtype);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Blogtype findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Blogtype blogtype, int pageNum,int pageSize);
	
}
